tests/e2e/specs/plugin-activation.test.js

```diff
@@ -34,37 +34,87 @@ const deactivateSiteKit = async () => {
 };
 
 describe( 'Plugin Activation Notice', () => {
-	beforeEach( async () => {
-		await deactivateSiteKit();
-		await activatePlugin( 'e2e-tests-gcp-credentials-plugin' );
-	} );
+	describe( 'When Javascript is enabled', () => {
+		beforeEach( async () => {
+			await deactivateSiteKit();
+			await activatePlugin( 'e2e-tests-gcp-credentials-plugin' );
+		} );
 
-	afterEach( async () => {
-		await deactivatePlugin( 'e2e-tests-gcp-credentials-plugin' );
-		await activateSiteKit();
-	} );
+		afterEach( async () => {
+			await deactivatePlugin( 'e2e-tests-gcp-credentials-plugin' );
+			await activateSiteKit();
+		} );
+
+		it( 'Should be displayed', async () => {
+			await activateSiteKit();
+
+			await page.waitForSelector( '.googlesitekit-activation__title' );
+
+			await expect( page ).toMatchElement( 'h3.googlesitekit-activation__title', { text: /Congratulations, the Site Kit plugin is now activated./i } );
+
+			await deactivateSiteKit();
+		} );
+
+		it( 'Should lead you to the setup wizard', async () => {
+			await activateSiteKit();
 
-	it( 'Should be displayed', async () => {
-		await activateSiteKit();
+			await page.waitForSelector( '.googlesitekit-activation' );
 
-		await page.waitForSelector( '.googlesitekit-activation__title' );
+			await expect( page ).toMatchElement( '.googlesitekit-start-setup', { text: 'Start setup' } );
 
-		await expect( page ).toMatchElement( 'h3.googlesitekit-activation__title', { text: 'Congratulations, the Site Kit plugin is now activated.' } );
+			await page.click( '.googlesitekit-start-setup' );
+			await page.waitForSelector( '.googlesitekit-wizard-step__title' );
 
-		await deactivateSiteKit();
+			// Ensure we're on the first step.
+			await expect( page ).toMatchElement( '.googlesitekit-wizard-progress-step__number--inprogress', { text: '1' } );
+
+			await deactivateSiteKit();
+		} );
+
+		it( 'Should not display noscript notice', async () => {
+			await activateSiteKit();
+
+			await expect( page ).not.toMatchElement( '.googlesitekit-noscript' );
+
+			await deactivateSiteKit();
+		} );
 	} );
 
-	it( 'Should lead you to the setup wizard', async () => {
-		await activateSiteKit();
+	describe( 'When Javascript is disabled', () => {
+		beforeEach( async () => {
+			await deactivateSiteKit();
+		} );
+
+		afterEach( async () => {
+			await activateSiteKit();
+		} );
+
+		it( 'Should not display plugin html', async () => {
+			// Disabling javascript beforeEach breaks utility functions.
```

Comment:
```suggestion
			// Disabling JavaScript in `beforeEach` breaks utility functions.
			// Each test without JavaScript must use
			// `await page.setJavaScriptEnabled( false );` and
			// `await page.setJavaScriptEnabled( true );` in the test itself.
```

Which functions does it break/mess with? Any extra context here would be good to know 😄